docs/advanced/refs.md

```diff
@@ -1,7 +1,59 @@
 # Refs
 *Refs* grant access to the actual Instance objects that are created by Roact. They're an escape hatch for when something is difficult or impossible to correctly express with the Roact API.
 
-To create a ref, pass a function prop with the key `Roact.Ref` when creating a primitive element:
+!!! info
+	Refs can only be used with primitive components.
+
+## Refs in Action
+
+To create a ref, pass a function prop with the key `Roact.Ref` when creating a primitive element.
+
+For example, suppose we wanted to create a search bar that captured cursor focus when any part of it was clicked. We might use a component like this:
+
+```lua
+--[[ A search bar with an icon and a text box that captures focus for 
+ its TextBox when its icon is clicked ]]
```

Comment:
As a minor style note (and something we should include in the style guide!), multi-line comments should generally start/end on their own line:

```lua
--[[
	A search bar with an icon and a text box that captures focus for its TextBox
	when its icon is clicked.
]]
```

Sublime's auto-wrap will fight you on this unless you explicitly select just the text before you press `Alt+Q`, which is kind of a shame.